src/main/scala/rx/lang/scala/Observable.scala

```diff
@@ -332,6 +332,143 @@ trait Observable[+T]
+   * Observable.
+   *
+   * Eager concatenation means that once a subscriber subscribes, this operator subscribes to all of the
+   * source [[Observable]]s. The operator buffers the values emitted by these [[Observable]]s and then drains them in
+   * order, each one after the previous one completes.
+   *
+   * ===Backpressure:===
+   * Backpressure is honored towards the downstream, however, due to the eagerness requirement, sources
+   * are subscribed to in unbounded mode and their values are queued up in an unbounded buffer.
+   *
+   * $noDefaultScheduler
+   *
+   * @param f the function that maps a sequence of values into a sequence of [[Observable]]s that will be
+   *          eagerly concatenated
+   * @return an [[Observable]] that emits items all of the items emitted by the [[Observable]]s returned by
+   *         `f`, one after the other, without interleaving them
+   * @since (if this graduates from Experimental/Beta to supported, replace this parenthetical with the release number)
+   */
+  @Experimental
+  def concatMapEager[R](f: T => Observable[R]): Observable[R] = {
+    toScalaObservable[R](asJavaObservable.concatMapEager[R](new Func1[T, rx.Observable[_ <: R]] {
+      def call(t1: T): rx.Observable[_ <: R] = {
+        f(t1).asJavaObservable
+      }
+    }))
+  }
+
+  /**
+   * $experimental Maps a sequence of values into [[Observable]]s and concatenates these [[Observable]]s eagerly into a single
+   * Observable.
+   *
+   * Eager concatenation means that once a subscriber subscribes, this operator subscribes to all of the
+   * source [[Observable]]s. The operator buffers the values emitted by these [[Observable]]s and then drains them in
+   * order, each one after the previous one completes.
+   *
+   * ===Backpressure:===
+   * Backpressure is honored towards the downstream, however, due to the eagerness requirement, sources
+   * are subscribed to in unbounded mode and their values are queued up in an unbounded buffer.
+   *
+   * $noDefaultScheduler
+   *
+   * @param f the function that maps a sequence of values into a sequence of [[Observable]]s that will be
+   *          eagerly concatenated
+   * @param capacityHint hints about the number of expected values in an [[Observable]]
+   * @return an [[Observable]] that emits items all of the items emitted by the [[Observable]]s returned by
+   *         `f`, one after the other, without interleaving them
+   * @since (if this graduates from Experimental/Beta to supported, replace this parenthetical with the release number)
+   */
+  @Experimental
+  def concatMapEager[R](f: T => Observable[R], capacityHint: Int): Observable[R] = {
+    toScalaObservable[R](asJavaObservable.concatMapEager[R](new Func1[T, rx.Observable[_ <: R]] {
+      def call(t1: T): rx.Observable[_ <: R] = {
+        f(t1).asJavaObservable
+      }
+    }, capacityHint))
+  }
```

Comment:
In `def flatMap[R](maxConcurrent: Int, f: T => Observable[R]): Observable[R]`, we have the integer as the first argument, and the function as the second argument. Any reason why to do it differently here? And do you know why they put the function first in RxJava?